src/app/pages/wallets/my-wallets/add-new-wallet/pages-wallets-my-wallets-add-new-wallet.component.ts

```diff
@@ -1,8 +1,34 @@
-import { Component } from '@angular/core';
+import { Component, EventEmitter, Output } from '@angular/core';
+import { MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { DialogCreateOrUpdateWalletComponent } from 'src/app/common/components/modals/dialog-create-or-update-wallet/dialog-create-or-update-wallet.component';
 
 @Component({
   selector: 'app-add-new-wallet',
   templateUrl: './pages-wallets-my-wallets-add-new-wallet.component.html',
   styleUrls: ['./pages-wallets-my-wallets-add-new-wallet.component.scss']
 })
-export class PagesWalletsMyWalletsAddNewWalletComponent {}
+export class PagesWalletsMyWalletsAddNewWalletComponent {
+
+    @Output('getWalletName') getWalletName = new EventEmitter<string>();
+
+    newWalletName: string = "";
+
+    constructor(
+        public dialog: MatDialog,
+    ) {}
+
+    openAddNewWalletDialog() {
```

Comment:
Like I said in comment below - you should not do this here but in parent component. Remember also about using fake request / gateway. Also please from the modal do not return just name of the wallet but object that will represent data for the wallet -> for now it should be just `{ name: string }` and we will extend it in the near future.